src/js/common/schemaform/SaveFormLink.jsx

```diff
@@ -28,35 +28,34 @@ class SaveFormLink extends React.Component {
       saveForm,
       savedStatus
     } = this.props;
-    let content = (<div>
-      <a onClick={this.openLoginModal}>Sign in before saving your application</a>
-      <LoginModal
-          key={1}
-          title="Sign in to save your application"
-          onClose={this.closeLoginModal}
-          visible={this.state.modalOpened}
-          user={this.props.user}
-          onUpdateLoginUrl={this.props.onUpdateLoginUrl}/>
-    </div>);
+    let content = <a onClick={saveForm}>Save and come back later</a>;
 
-    if (this.props.user.login.currentlyLoggedIn) {
-      content = <a onClick={saveForm}>Save and come back later</a>;
+    if (!this.props.user.login.currentlyLoggedIn) {
+      // if we have a noAuth status, that means they tried to save and got a 401, which
+      // likely means their session is expired (since they were logged in before)
+      content = (<div>
+        {savedStatus === SAVE_STATUSES.noAuth
+            ? <span>Sorry, your session has expired. <a onClick={this.openLoginModal}>Please sign in again</a></span>
+            : <a onClick={this.openLoginModal}>Sign in before saving your application</a>}
```

Comment:
@stacyhlinn What do you think about having just "Sign in" linked here too? It would make it consistent with the intro page and session expiration (once that's updated per your comment).

Actually, in your design, this just said "Save and come back later" both when the user is and is not logged in. Is that what we want to go with?